iothub/device/src/Transport/TransportHandler.cs

```diff
@@ -25,6 +26,7 @@ public override Task<Message> ReceiveAsync(CancellationToken cancellationToken)
 
         public override Task WaitForTransportClosedAsync()
         {
+            _transportShouldRetry = new TaskCompletionSource<bool>();
```

Comment:
Coding standards: please move back the initialization next to the definition.

Reply:
That's not the case. Every time  `WaitForTransportClosedAsync`, it should create a new instance.

Reply:
As per above, this is by design. We are reusing transport handler and hence cannot move it back.

Reply:
Please refer to the design: the transport handler must be disposed on retry. Otherwise we need to also change Mqtt and Http handlers since this is common code.